File: argmin.py
from optimizers import sgd,adam

def argmin(U,dU, x0):
    def callback(params, t, g):
        if U(params, t) < 1e-3:
            return True
        return False
    x = adam(dU, x0, step_size = .01,num_iters=1000, callback=callback)
    return x

File: argmin.m
function x = argmin(U,dU, x0)
  while 1
    x = x0 - 0.01*dU(x0);
    if abs(U(x0)-U(x))< 1e-9
      break;
    end
    x0 = x;
  end
end
